### ISSUE TITLE
[Bug]: V9 Buttons in High Contrast do not meet contrast on hover

### ISSUE THREAD

**Issue by SavannahBourgeois** [microsoft/fluentui#23340]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
Repros with windows 10 and 11. Edge Version 101.0.1210.47 (Official build) (64-bit)
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://codesandbox.io/s/qgljnr?file=/example.tsx

## Actual Behavior
Hover state does not meet contrast behavior in high contrast black #1 theme in windows 10. Screenshot below shows a button on hover in High Contast Black #1 and the text does not have enough of a contrast with the background

![image](https://user-images.githubusercontent.com/42548121/171281594-d584004b-8fa2-4d99-8fde-aabba08c5c6a.png)

## Expected Behavior
Hover states meets contrast ratio especially in high contrast themes.

### Requested priority

Normal

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by SavannahBourgeois** [comment]
@khmakoto logged an issue so it wouldn't get lost
